File: 1679-shortest-subarray-to-be-removed-to-make-array-sorted/shortest-subarray-to-be-removed-to-make-array-sorted.js
var findLengthOfShortestSubarray = function(arr) {
    let N = arr.length
    let r = N - 1
    let l = 0
    let output = N 

    while (r > 0 && arr[r - 1] <= arr[r]) r -= 1    
    while (l + 1 <= N && arr[l] <= arr[l + 1]) l += 1

    output = r
    if (l == N - 1) return 0
    output = Math.min(r, N - l - 1, output)


    let i = 0
    let j = r

    while (i <= l && j < N) {
        if (arr[i] <= arr[j]) {
            length = j - i - 1
            output = Math.min(output, length)
            i += 1
        }
        else {
            j ++
        }
    }

    return output
};

File: 1679-shortest-subarray-to-be-removed-to-make-array-sorted/shortest-subarray-to-be-removed-to-make-array-sorted.py
class Solution:
    def findLengthOfShortestSubarray(self, arr: List[int]) -> int:
        N = len(arr)
        i = 0
        j = N - 1
        output = N

        while (i < N -1 and arr[i] <= arr[i + 1]):
            i += 1

        while (j > 0 and arr[j] >= arr[j - 1]):
            j -=1 

        if (j == 0):
            return 0

        output = min(output, j, N - i - 1)
        l = 0
        r = j

        while (l <= i and r < N):
            if (arr[l] <= arr[r]):
                output = min(output, r- l - 1)
                l += 1
            else:
                r += 1

        return output 
        
        